### PR TITLE
Update EE repositories/remotes/registries endpoints for new API

### PR DESCRIPTION
**Pull request by jerabekjiri**
Issue: [AAH-1721](https://issues.redhat.com/browse/AAH-1721)

Depends on [#1397](https://github.com/ansible/galaxy_ng/pull/1397)

- unify EE endpoints to expose `id` (~~pulp_id~~) and `pulp_href` 
- move to pulp endpoints for rbac assignment in execution environment owners tab
- refactor `owners-tab` component to work with pulp_container/namespaces/: list_roles, add_role, remove_role, my_permissions endpoints
 

- galaxykit PR: [#64](https://github.com/ansible/galaxykit/pull/64)

### CODE REVIEW

**Comment by himdel** [comment]
The test failures look relevant, is this just because the API is not merged yet?

I guess you could do something like... for now.

```
diff --git a/.github/workflows/cypress.yml b/.github/workflows/cypress.yml
index f340f3cf..59b1cbb4 100644
--- a/.github/workflows/cypress.yml
+++ b/.github/workflows/cypress.yml
@@ -37,6 +37,9 @@ jobs:
         # blow up without galaxy_ng commit info
         [ -n "$GALAXY_NG_COMMIT" ]
 
+        SHORT_BRANCH=fix/refactor-ee-repositories
+        GALAXY_NG_COMMIT=234f46b54ee86cc92be03ea0e09375eca85b0d53
+
         echo "SHORT_BRANCH=${SHORT_BRANCH}" >> $GITHUB_ENV
         echo "GALAXY_NG_COMMIT=${GALAXY_NG_COMMIT}" >> $GITHUB_ENV
 
@@ -71,7 +74,7 @@ jobs:
 
           RUN pip3 install --upgrade \
             requests \
-            git+https://github.com/ansible/galaxy_ng.git@${{ env.SHORT_BRANCH }}
+            git+https://github.com/jerabekjiri/galaxy_ng.git@${{ env.SHORT_BRANCH }}
 
           RUN mkdir -p /etc/nginx/pulp/
           RUN ln /usr/local/lib/python*/site-packages/pulp_ansible/app/webserver_snippets/nginx.conf /etc/nginx/pulp/pulp_ansible.conf
```

**Review by newswangerd — APPROVED** [review]

File: src/api/execution-environment-namespace.ts [hunk 1]
```diff
@@ -1,7 +1,23 @@
 import { HubAPI } from './hub';
 
 class API extends HubAPI {
-  apiPath = this.getUIPath('execution-environments/namespaces/');
+  apiPath = 'pulp/api/v3/pulp_container/namespaces/';
```
[the remark this answers is not in the document]
Comment (jerabekjiri) [reply src/api/execution-environment-namespace.ts]: Fixed, thanks :)

**Comment by jerabekjiri** [comment]
> The test failures look relevant, is this just because the API is not merged yet?
> 
> I guess you could do something like... for now.
> 
> ```diff
> diff --git a/.github/workflows/cypress.yml b/.github/workflows/cypress.yml
> index f340f3cf..59b1cbb4 100644
> --- a/.github/workflows/cypress.yml
> +++ b/.github/workflows/cypress.yml
> @@ -37,6 +37,9 @@ jobs:
>          # blow up without galaxy_ng commit info
>          [ -n "$GALAXY_NG_COMMIT" ]
>  
> +        SHORT_BRANCH=fix/refactor-ee-repositories
> +        GALAXY_NG_COMMIT=234f46b54ee86cc92be03ea0e09375eca85b0d53
> +
>          echo "SHORT_BRANCH=${SHORT_BRANCH}" >> $GITHUB_ENV
>          echo "GALAXY_NG_COMMIT=${GALAXY_NG_COMMIT}" >> $GITHUB_ENV
>  
> @@ -71,7 +74,7 @@ jobs:
>  
>            RUN pip3 install --upgrade \
>              requests \
> -            git+https://github.com/ansible/galaxy_ng.git@${{ env.SHORT_BRANCH }}
> +            git+https://github.com/jerabekjiri/galaxy_ng.git@${{ env.SHORT_BRANCH }}
>  
>            RUN mkdir -p /etc/nginx/pulp/
>            RUN ln /usr/local/lib/python*/site-packages/pulp_ansible/app/webserver_snippets/nginx.conf /etc/nginx/pulp/pulp_ansible.conf
> ```

very cool :+1:

**Comment by himdel** [comment]
On EE *list* screen, try to Edit a remote or local EE, Save:
404 - PATCH /api/automation-hub/pulp/api/v3/distributions/container/container/**undefined**/

on the detail screen, this works, but on the list screen the id is undefined now

Other than that, I think everything works :+1: 
(Do remember to remove the cypress.yml change before merging :))

**Comment by jerabekjiri** [comment]
> * unify EE endpoints to expose `id` (~pulp_id~) and `pulp_href`
> *

oops, nice catch, I didn't notice we use `ContainerDistributionAPI.patch` when the description changes

**Review by himdel — APPROVED** [review]

**Pull request merged**